recipe/build-pyarrow.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/sh
+
+set -e
+set -x
+
+# Build dependencies
+export ARROW_HOME=$PREFIX
+export PARQUET_HOME=$PREFIX
+export SETUPTOOLS_SCM_PRETEND_VERSION=$PKG_VERSION
+export PYARROW_BUILD_TYPE=release
+export PYARROW_WITH_DATASET=1
+export PYARROW_WITH_FLIGHT=1
+export PYARROW_WITH_GANDIVA=1
+export PYARROW_WITH_HDFS=1
+export PYARROW_WITH_ORC=1
+export PYARROW_WITH_PARQUET=1
+export PYARROW_WITH_PLASMA=1
+export PYARROW_WITH_S3=1
+BUILD_EXT_FLAGS=""
+
+# Enable CUDA support
+if [[ ! -z "${cuda_compiler_version+x}" && "${cuda_compiler_version}" != "None" ]]
+then
+    export PYARROW_WITH_CUDA=1
+    BUILD_EXT_FLAGS="${BUILD_EXT_FLAGS} --with-cuda"
+else
+    export PYARROW_WITH_CUDA=0
+fi
+
+# Resolve: Make Error at cmake_modules/SetupCxxFlags.cmake:338 (message): Unsupported arch flag: -march=.
+if [[ "$(uname -m)" = "aarch64" ]]
+then
+    export PYARROW_CMAKE_OPTIONS="-DARROW_ARMV8_ARCH=armv8-a"
+fi
+
+cd python
+
+$PYTHON setup.py \
+        build_ext $BUILD_EXT_FLAGS \
+        install --single-version-externally-managed \
+                --record=record.txt
```

Comment:
Are we able to use `pip` here?

Reply:
We should check that but in a separate PR.

Reply:
Yeah that makes sense. Was having a little trouble seeing how this was built before. If this is no different from what we were already doing, agree we don't need to change it here.

Reply:
Using pip does not work out of the box:
```
$PYTHON -m pip install . --no-deps -vv build_ext $BUILD_EXT_FLAGS install
```
fails with
```
pip._internal.exceptions.DistributionNotFound: No matching distribution found for build_ext
```

Reply:
You need something like,
```
$PYTHON -m pip install . --no-deps -vv --global-option=build_ext --global-option=--with-cuda 
```

Reply:
Right. The following diff works locally:
```
diff --git a/recipe/build-pyarrow.sh b/recipe/build-pyarrow.sh
index 99396d8..4b01871 100644
--- a/recipe/build-pyarrow.sh
+++ b/recipe/build-pyarrow.sh
@@ -22,7 +22,7 @@ BUILD_EXT_FLAGS=""
 if [[ ! -z "${cuda_compiler_version+x}" && "${cuda_compiler_version}" != "None" ]]
 then
     export PYARROW_WITH_CUDA=1
-    BUILD_EXT_FLAGS="${BUILD_EXT_FLAGS} --with-cuda"
+    BUILD_EXT_FLAGS="${BUILD_EXT_FLAGS} --global-option=build_ext --global-option=--with-cuda"
 else
     export PYARROW_WITH_CUDA=0
 fi
@@ -35,7 +35,4 @@ fi
 
 cd python
 
-$PYTHON setup.py \
-        build_ext $BUILD_EXT_FLAGS \
-        install --single-version-externally-managed \
-                --record=record.txt
+$PYTHON -m pip install . --no-deps -vv $BUILD_EXT_FLAGS
diff --git a/recipe/meta.yaml b/recipe/meta.yaml
index 8a0edf7..20523e2 100644
--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -181,6 +181,7 @@ outputs:
         - boost-cpp
         - cython
         - numpy 1.16.*
+        - pip
         - python
         - setuptools
         - setuptools_scm
```
Just FYI.